Sync/HistoryPayload.swift

```diff
@@ -17,15 +19,19 @@ open class HistoryPayload: CleartextPayloadJSON {
     }
 
     override open func isValid() -> Bool {
+        log.debug("Calling HistoryPayload.isValid()...")
         if !super.isValid() {
+            log.debug("HistoryPayload.isValid() -> false")
             return false
         }
 
         if self["deleted"].bool ?? false {
+            log.debug("HistoryPayload.isValid() -> true")
             return true
         }
 
-        return self["histUri"].string != nil &&      // TODO: validate URI.
+        log.debug("HistoryPayload.isValid() -> " + ((self["histUri"].string != nil && self["title"].isStringOrNull() && self["visits"].isArray()) ? "true" : "false"))
```

Comment:
Given that this is the risky format, split this up into three lines, and also log the ID.